Recruit-Training/Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,23 +20,48 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return totalAmphoursUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    return (current_ < 0);
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    double totalHoursRemaining;
+
+    if(isCharging())
+    {
+        totalHoursRemaining = abs(totalAmpHoursUsed() / current_);
+    }
+    else
+    {
+        totalHoursRemaining = (BATTERY_AMP_HOUR_CAPACITY - totalAmpHoursUsed()) / current_;
+    }
+
+    QTime time(0, 0, 0, 0);
```

Comment:
You can make this `timeRemaining`, and then reuse it later with `timeRemaining = timeRemaining.addMSecs(..)`